testsuite/MDAnalysisTests/analysis/test_contacts.py

```diff
@@ -171,16 +171,61 @@ class TestContacts(object):
     def universe():
         return mda.Universe(PSF, DCD)
 
-    def _run_Contacts(self, universe, start=None, stop=None, step=None, **kwargs):
+    def _run_Contacts(
+        self, universe,
+        ag_acidic=None, ag_basic=None,
+        start=None, stop=None, step=None, **kwargs
+    ):
         acidic = universe.select_atoms(self.sel_acidic)
         basic = universe.select_atoms(self.sel_basic)
+        if ag_acidic is None:
+            ag_acidic = self.sel_acidic
+        if ag_basic is None:
+            ag_basic = self.sel_basic
+
         return contacts.Contacts(
             universe,
-            select=(self.sel_acidic, self.sel_basic),
+            select=(ag_acidic, ag_basic),
             refgroup=(acidic, basic),
             radius=6.0,
             **kwargs).run(start=start, stop=stop, step=step)
 
+    def test_select_valid_types(self, universe):
+        """Test if Contact can take both string and AtomGroup as selections.
+        """
+        acidic = universe.select_atoms(self.sel_acidic)
+        basic = universe.select_atoms(self.sel_basic)
+
+        with_string = self._run_Contacts(universe)
+        with_atomgroup = self._run_Contacts(
+            universe, ag_acidic=acidic, ag_basic=basic
+        )
+
+        assert_equal(with_atomgroup.grA, with_string.grA)
+        assert_equal(with_atomgroup.grB, with_string.grB)
+
+    @pytest.mark.parametrize(
+        "ag1, ag2", 
+        [
+            (1, 2),
+            ([1], [2]),
+            (mda.Universe, sel_basic),
+            (sel_acidic, 2),
+            ("USE UPDATING AG", sel_basic),
+            (sel_acidic, "USE UPDATING AG")
+        ]
+    )
+    def test_select_wrong_types(self, universe, ag1, ag2):
+        """test that Contacts fails if selection types are wrong"""
+        if ag1 == "USE UPDATING AG":
+            ag1 = universe.select_atoms(self.sel_acidic, updating=True)
+        if ag2 == "USE UPDATING AG":
+            ag2 = universe.select_atoms(self.sel_basic, updating=True)
+
+        sel_err_msg = contacts.Contacts.select_error_message
```

Comment:
No need to duplicate here, can just use `contacts.Contacts.select_error_message` directly in the `pytest.raises expression`